src/instructlab/model/chat.py

```diff
@@ -648,6 +652,13 @@ def start_prompt(
 
         self.log_message(PROMPT_PREFIX + content + "\n\n")
 
+        # if RAG is enabled, fetch context and insert into session
+        # TODO: what if context is already too long? note that current retriever implementation concatenates all docs
+        # TODO: better way to check whether we should perform retrieval?
```

Comment:
why not using `ctx.obj.config.chat.rag.enabled` or an internal method hiding the same logic?

Reply:
`ctx` is not available here. `retriever` should be non-`None` only if that config parameter is enabled, so the `enabled` check does happen farther up.